contrib/datadog/tracing/interceptor.go

```diff
@@ -46,6 +46,9 @@ type TracerOptions struct {
 
 	// DisableQueryTracing can be set to disable query tracing.
 	DisableQueryTracing bool
+
+	// ErrCheckFn can be set to a custom function to determine if an error should be traced.
+	ErrCheckFn func(err error) bool
```

Comment:
`Fn` not an abbreviation we usually use. How about:

```suggestion
	// CheckError can be set to a custom function to determine if an error should be traced.
	CheckError func(error) bool
```

Or maybe even better and more flexible:

```suggestion
	// OnFinish sets finish options. If unset, this will use [tracer.WithError] if 
	// [interceptor.TracerFinishSpanOptions.Error] is non-nil and not
	// [workflow.IsContinueAsNewError].
	OnFinish func(*interceptor.TracerFinishSpanOptions) []tracer.FinishOption
```

@Quinn-With-Two-Ns, preference?

Reply:
Updated the naming, will be waiting for the final decision regarding the more flexible option.

Reply:
+1 to using `TracerFinishSpanOptions` to allow us to add more fields in the future.

Reply:
Implemented 👌